### PR TITLE
chore(Card): convert Card react demos to TS

### LINKED ISSUE #9374: Convert card react demos to TS
Components -> Card -> React demos

The following need TS conversions:

- Horizontal card grid
- Horizontal split
- Details card
- Utilization card 1
- Utilization card 2
- Utilization card 4
- Nested cards
- Trend card 2
- Log view
- Events view

### PR DESCRIPTION
**Pull request by jenny-s51**
**What**: Closes #9374 

Converted remaining Card react demos to TS and fixed various linting / type errors

### CODE REVIEW

File: packages/react-core/src/demos/CardDemos.md [hunk 1]
```diff
@@ -2031,7 +2046,7 @@ import {
   Flex,
   FlexItem
 } from '@patternfly/react-core';
-import { ChartArea, ChartContainer, ChartGroup, ChartLabel, ChartVoronoiContainer } from '@patternfly/react-charts';
+import { ChartArea, ChartGroup, ChartVoronoiContainer } from '@patternfly/react-charts';
 
 const AccordionCard: React.FunctionComponent = () => {
```
Comment (thatblindgeye) [inline packages/react-core/src/demos/CardDemos.md:2051]:
```suggestion
const CardWithAccordion: React.FunctionComponent = () => {
```

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Some comments/pretty quick updates below. Could we also create separate files for these demos in a `demos/examples/CardDemos` directory?

File: packages/react-core/src/demos/CardDemos.md [hunk 2]
```diff
@@ -40,281 +40,269 @@ import {
...
-};
+const CardHorizontalSplitDemo: React.FunctionComponent = () => (
+  <Card id="card-demo-horizontal-split-example" isFlat>
+    <Grid md={6}>
+      <GridItem
+        style={{
+          minHeight: '200px',
+          backgroundPosition: 'center',
+          backgroundSize: 'cover',
+          backgroundImage: 'url(/assets/images/pfbg_992@2x.jpg)'
```
[the remark this answers is not in the document]
Comment (jenny-s51) [reply packages/react-core/src/demos/CardDemos.md:283]: Thanks, nice catch @thatblindgeye - this needs an absolute url, will update

File: packages/react-core/src/demos/CardDemos.md [hunk 3]
```diff
@@ -40,281 +40,269 @@ import {
...
+        toggleButtonProps={{
+          id: 'toggle-button',
+          'aria-label': 'Actions',
+          'aria-labelledby': 'titleId toggle-button',
+          'aria-expanded': isCardExpanded
+        }}
+      >
+        {isCardExpanded && <CardTitle id="titleId">Getting Started</CardTitle>}
+        {!isCardExpanded && (
+          <Level hasGutter>
```
[the remark this answers is not in the document]
Comment (jenny-s51) [reply packages/react-core/src/demos/CardDemos.md:113]: Great point @thatblindgeye - I think technically this is out of scope but would be a nice discussion to have. Could be worth tracking this in a separate issue

File: packages/react-core/src/demos/CardDemos.md [hunk 4]
```diff
@@ -955,7 +964,7 @@ const descriptionListData = [
 
 const Status: React.FunctionComponent = () => {
   const [activeTabKey, setActiveTabKey] = React.useState(0);
-  const handleTabClick = (event: React.MouseEvent, tabIndex: number) => {
+  const handleTabClick = (tabIndex: number) => {
```
[the remark this answers is not in the document]
Comment (jenny-s51) [reply packages/react-core/src/demos/CardDemos.md]: Yes, good idea 👍 updated

**Review by tlabaj — APPROVED** [review]
LGTM. Just needs a rebase.

**Pull request merged**